src/engine/engine_common.c

```diff
@@ -122,36 +147,38 @@ void ocf_engine_update_req_info(struct ocf_cache *cache,
 				start_sector, end_sector))
 				req->info.dirty_all++;
 		}
-
-		if (req->part_id != ocf_metadata_get_partition_id(cache,
-				_entry->coll_idx)) {
-			/*
-			 * Need to move this cache line into other partition
-			 */
-			_entry->re_part = true;
-			req->info.re_part_no++;
-		}
-
 		break;
 	case LOOKUP_MISS:
-		req->info.seq_req = false;
 		break;
 	case LOOKUP_INSERTED:
 	case LOOKUP_REMAPPED:
+		req->info.insert_no++;
 		break;
 	default:
 		ENV_BUG();
 		break;
 	}
 
-	/* Check if cache hit is sequential */
-	if (req->info.seq_req && entry) {
-		if (ocf_metadata_map_lg2phy(cache,
-			(req->map[entry - 1].coll_idx)) + 1 !=
-			ocf_metadata_map_lg2phy(cache,
-			_entry->coll_idx)) {
-			req->info.seq_req = false;
-		}
+	if ((status == LOOKUP_HIT || status == LOOKUP_REMAPPED) &&
+			req->part_id != ocf_metadata_get_partition_id(cache,
+					_entry->coll_idx)) {
+		/*
+		 * Need to move this cache line into other partition
+		 */
+		_entry->re_part = true;
+		req->info.re_part_no++;
+	}
+
+	if (status == LOOKUP_MISS)
+		return;
+
+	/* Check if cacheline is sequential with previous */
+	if (entry > 0 && ocf_engine_clines_phys_cont(req, entry - 1))
+		req->info.seq_no++;
+	/* Check if cacheline is sequential with next */
+	if (entry < req->core_line_count - 1 &&
+			ocf_engine_clines_phys_cont(req, entry)) {
+		req->info.seq_no++;
```

Comment:
I don't get it - you're incrementing `seq_no` up to twice per map entry. This way if you have 3 sequential core lines in request you'd get `seq_no` equal to 4, which makes no sense. Is there something I'm missing?

Reply:
not twice, during initial traversation entry2->status will always be 0, so each execution of this piece of code will increment the counter at most once.

Reply:
However now I see that I didn't consider the fact, that the tracersation itself is called twice, so the counter might be incremented too many times.

Reply:
Now I see a problem with my implementation - traversation is performed twice (initial and before eviction). map->status is not zeroed before each traversation. So there will be stale status in map[entry+1] during the second traversation.

Reply:
Ok, I am adjusting the code so that this "double check" (for previous and and next cacheline being sequential with current) is performed explicitly only for cachelines that are "patched" as a result of insertion via eviction. So the existing traversation, mapping and "update_req" functions would have minimal changes in sequential detection logic. 
And eviction code would need to additionally call "patch_req" to update stats for the remapped cacheline. Also there would be no checks for "entry->status != 0", as these were not needed in the baseline and now patch_req() is always called with request fully traversed, meaning all the cachelines have a proper tatus set.